pkg/clustermesh/kvstoremesh/remote_cluster.go

```diff
@@ -254,14 +254,7 @@ func (rc *remoteCluster) waitForConnection(ctx context.Context) {
 	case <-rc.synced.connected:
 	case <-time.After(rc.readyTimeout):
 		rc.logger.Info("Remote cluster did not connect within timeout, removing from readiness checks")
-		for {
-			select {
-			case <-rc.synced.resources.WaitChannel():
-				return
-			default:
-				rc.synced.resources.Done()
-			}
-		}
+		rc.synced.resources.Release()
```

Comment:
Essentially I moved the above loop "as-is" into the new `StoppableWaitGroup.Release()` method. Might use a better name, or alternatively we don't add it and instead call the done funcs of each of the resources one by one here. Thoughts?

Reply:
I'd personally just directly call the individual done functions here. `Release` looks a bit dangerous to me, considering that it will lead to a busy loop if the SWG has not been stopped yet (which is never the case here, but may lead to surprises in the future).

Reply:
Done!